contracts/SimpleYieldEulerLever.sol

```diff
@@ -0,0 +1,395 @@
+    ///     level functionality is as follows:
+    ///         - We have supplied and borrowed underlying asset.
+    ///         - We deposit it to euler and put the etoken received in the vault.
+    ///         - Against it, we borrow enough fyToken to sell & repay the flash loan.
+    /// @param poolAddress The pool (and thereby series) to borrow from.
+    /// @param vaultId The vault id to put collateral into and borrow from.
+    /// @param borrowAmount The amount of underlying asset borrowed in the flash loan.
+    /// @param fee The fee that will be issued by the flash loan.
+    /// @param baseAmount The amount of own collateral to supply.
+    /// @param minCollateral The final amount of collateral to end up with, or
+    ///     the function will revert. Used to prevent slippage.
+    function borrow(
+        bytes12 vaultId,
+        address poolAddress,
+        uint256 borrowAmount,
+        uint256 fee,
+        uint256 baseAmount,
+        uint256 minCollateral
+    ) internal {
+        // Deposit to get Euler token in return which would be used to payback flashloan
+        eToken.deposit(0, borrowAmount - fee);
+
+        uint256 eBalance = IERC20(address(eToken)).balanceOf(address(this));
+
+        IERC20(address(eToken)).transfer(ilkJoin, eBalance);
+
+        _pourAndSell(vaultId, eBalance, borrowAmount, poolAddress);
+    }
+
+    /// @dev Additional function to get over stack too deep
+    /// @param vaultId VaultId
+    /// @param baseAmount Amount of collateral
+    /// @param borrowAmount Amount being borrowed
+    /// @param poolAddress Address of the pool to trade on
+    function _pourAndSell(
+        bytes12 vaultId,
+        uint256 baseAmount,
+        uint256 borrowAmount,
+        address poolAddress
+    ) internal {
+        IPool pool = IPool(poolAddress);
+        uint128 fyBorrow = pool.buyBasePreview(borrowAmount.u128());
+        ladle.pour(
+            vaultId,
+            address(pool),
+            baseAmount.u128().i128(),
+            fyBorrow.i128()
+        );
+        pool.buyBase(address(this), borrowAmount.u128(), fyBorrow);
+    }
+
+    /// @notice divest a position.
+    ///
+    ///     If pre maturity, borrow liquidity tokens to repay `art` debt and
+    ///     take `ink` collateral.
+    ///
+    ///     If post maturity, borrow USDC/DAI to pay off the debt directly.
+    ///
+    ///     This function will take the vault from you using `Giver`, so make
+    ///     sure you have given it permission to do that.
+    /// @param seriesId The seriesId corresponding to the vault.
+    /// @param vaultId The vault to use.
+    /// @param ink The amount of collateral to recover.
+    /// @param art The debt to repay.
+    /// @dev It is more gas efficient to let the user supply the `seriesId`,
+    ///     but it should match the pool.
+    function divest(
+        bytes6 seriesId,
+        bytes12 vaultId,
+        uint256 ink,
+        uint256 art
```

Comment:
Please make `ink` and `art` uint128, as for `invest`. More consistent, and simpler.

Alternatively, make all input parameters uint256 across the whole contract, and convert them to other types only when storing them or passing them to other contracts that require less than uint256 as parameters.

Either option is fine with me.